libraries/docker_container.rb

```diff
@@ -32,56 +32,56 @@ class DockerContainer < DockerBase
     property :repo, String, default: lazy { container_name }
     property :tag, String, default: 'latest'
     property :command, ShellCommand
-    property :attach_stderr, Boolean, desired_state: false
-    property :attach_stdin, Boolean, desired_state: false
-    property :attach_stdout, Boolean, desired_state: false
+    property :attach_stderr, Boolean, default: false, desired_state: false
+    property :attach_stdin, Boolean, default: false, desired_state: false
+    property :attach_stdout, Boolean, default: false, desired_state: false
     property :autoremove, Boolean, desired_state: false
-    property :cap_add, NonEmptyArray
-    property :cap_drop, NonEmptyArray
+    property :cap_add, NonEmptyArray, default: nil
+    property :cap_drop, NonEmptyArray, default: nil
     property :cgroup_parent, String, default: ''
     property :cpu_shares, [Fixnum, nil], default: 0
     property :cpuset_cpus, String, default: ''
     property :detach, Boolean, default: true, desired_state: false
-    property :devices, ArrayType
-    property :dns, NonEmptyArray
-    property :dns_search, ArrayType
+    property :devices, Array, default: []
+    property :dns, Array, default: []
```

Comment:
because ArrayType will flip them to nil if it is an empty Array, but docker defaults them to being an empty array [].